Makefile.am

```diff
@@ -90,3 +90,11 @@ test_test_scrypt_SOURCES =	test/test_scrypt.c			\
 			lib/crypto/crypto_scrypt_smix.c			\
 			cpusupport-config.h
 test_test_scrypt_LDADD=	libscrypt_sse2.a
+
+.PHONY: test
+# we can't only build "scrypt test/test_scrypt" because that won't build
+# the BUILT_SOURCES.
```

Comment:
On Sun, Mar 13, 2016 at 01:10:07PM -0700, Colin Percival wrote:
>    Never mind, I think I understand now. autotools is just being stupid and
>    not realizing what it means for a .h file to be a source file.

Yeah.  :(

One attractive option (to me) is to rename it to cpusupport-config.c.
According to the automake manual, this should work much better because it
already knows about .c files being dependencies.  That way we don't need to
mess with BUILT_SOURCES at all.